R/clone.R

```diff
@@ -89,7 +89,11 @@ generator_funs$clone_method <- function(deep = FALSE) {
       # fields that are R6 objects.
       deep_clone <- function(name, value) {
         # Check if it's an R6 object.
-        if (is.environment(value) && !is.null(value$`.__enclos_env__`)) {
+        if (
+          is.environment(value) &&
+          ".__enclos_env__" %in% ls(value, all.names = TRUE) &&
+          !is.null(get(".__enclos_env__", value))
+        ) {
```

Comment:
Maybe assign the complex conditional expression to a variable? 
Makes the code easier to read and also introduces a reusable abstraction.

```suggestion
        is_r6_object <- is.environment(value) &&
          ".__enclos_env__" %in% ls(value, all.names = TRUE) &&
          !is.null(get(".__enclos_env__", value))   
        if (is_r6_object) {
```

Reply:
While applying your suggestion I realized `ls(value, all.names = TRUE)` had a huge performance impact (the check was at least one order of magnitude slower, and it did not scale great with the number of elements in `value` for our use case).

So I tried several alternatives (`ls`, `names`, `rlang::env_get`) without success until I finally found `get0()` which is in base R since R 3.2 and it is just what we needed.

Thanks for your suggestion, it helped me realize of this other problem and address it.

I believe the patch can now safely be merged.